machinist/mserver/controller.go

```diff
@@ -34,7 +45,11 @@ func (en *Controller) HandlePing(stream machinist.Controller_PollServer, ping *m
 
 func (en *Controller) HandleRegister(stream machinist.Controller_PollServer, ping *machinist.ClientRegister) error {
 	en.Lock()
-	en.ConnectedNodes[ping.Name] = ping.Tag
+	n := &Node{
+		Name: ping.Name,
+		Tags: ping.Tag,
+	}
+	en.connectedNodes[ping.Name] = n
```

Comment:
nit:
```
n := &Node{
...
}
en.Lock()
en.connectedNodes[ping.Name] = n
en.Unlock()
```

better even would be to have an "Add" method on the Controller object? or have a locked connectedNodes object.
That way you could use defer (which is panic safe)